includes/Core/Consent_Mode/REST_Consent_Mode_Controller.php

```diff
@@ -188,36 +188,35 @@ protected function get_rest_routes() {
 					array(
 						'methods'             => WP_REST_Server::READABLE,
 						'callback'            => function () {
-							$is_active  = function_exists( 'wp_set_consent' );
-							$installed  = $is_active;
-							$plugin_uri = 'https://wordpress.org/plugins/wp-consent-api';
-							$plugin     = 'wp-consent-api/wp-consent-api.php';
-
-							$response = array(
+							// Here we intentionally use a non-plugin-specific detection strategy.
+							$is_active = function_exists( 'wp_set_consent' );
+							$response  = array(
 								'hasConsentAPI' => $is_active,
 							);
 
+							// Alternate wp_nonce_url without esc_html breaking query parameters.
+							$nonce_url = function ( $action_url, $action ) {
+								return add_query_arg( '_wpnonce', wp_create_nonce( $action ), $action_url );
+							};
+
 							if ( ! $is_active ) {
-								$consent_mode_plugin_status = Plugin_Status::get_plugin_status( $plugin, $plugin_uri );
+								$installed_plugin = $this->get_consent_api_plugin_file();
 
-								if ( Plugin_Status::PLUGIN_STATUS_INSTALLED === $consent_mode_plugin_status ) {
-									$installed = true;
-								}
+								$consent_plugin = array(
+									'installed'   => (bool) $installed_plugin,
+									'installURL'  => false,
+									'activateURL' => false,
+								);
 
-								// Alternate wp_nonce_url without esc_html breaking query parameters.
-								$nonce_url = function ( $action_url, $action ) {
-									return add_query_arg( '_wpnonce', wp_create_nonce( $action ), $action_url );
-								};
+								if ( ! $installed_plugin && current_user_can( 'install_plugins' ) ) {
+									$consent_plugin['installURL'] = $nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=wp-consent-api' ), 'install-plugin_wp-consent-api' );
+								}
 
-								$plugin       = Plugin_Status::$plugin_path;
-								$activate_url = $nonce_url( self_admin_url( 'plugins.php?action=activate&plugin=' . $plugin ), 'activate-plugin_' . $plugin );
-								$install_url  = $nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=wp-consent-api' ), 'install-plugin_wp-consent-api' );
+								if ( $installed_plugin && current_user_can( 'activate_plugin', $installed_plugin ) ) {
+									$consent_plugin['activateURL'] = $nonce_url( self_admin_url( 'plugins.php?action=activate&plugin=' . $installed_plugin ), 'activate-plugin_' . $installed_plugin );
+								}
 
-								$response['wpConsentPlugin'] = array(
-									'installed'   => $installed,
-									'activateURL' => current_user_can( 'activate_plugin', $plugin ) ? esc_url_raw( $activate_url ) : false,
-									'installURL'  => current_user_can( 'install_plugins' ) ? esc_url_raw( $install_url ) : false,
-								);
+								$response['wpConsentPlugin'] = $consent_plugin;
```

Comment:
This cleans up a few things by extracting conditions that were previously inline/ternary but also only providing an activate URL if the plugin is installed; Similarly only providing an install URL if the plugin isn't installed.